Sources/Controllers/Cells/OAPoiTableViewCell.m

```diff
@@ -180,18 +195,27 @@ - (void) collectionView:(UICollectionView *)collectionView didSelectItemAtIndexP
     if (collectionView.tag == kCategoryCellIndex)
     {
         NSDictionary *item = _categoryDataArray[indexPath.row];
+        _currentCategoryIndex = indexPath.row;
         _currentCategory = item[@"categoryName"];
+        
+        [self.categoriesCollectionView layoutIfNeeded];
         [self.categoriesCollectionView reloadData];
+        
+        [self.collectionView layoutIfNeeded];
         [self.collectionView reloadData];
+        
+        [self layoutIfNeeded];
+        [self layoutSubviews];
```

Comment:
Prefer using setNeedsLayout/layoutIfNeeded combination. It is a very poor practice to call layoutSubviews directly